nanolog_test.go

```diff
@@ -48,7 +48,7 @@ func TestSetWriter(t *testing.T) {
 	SetWriter(buf)
 
 	// simulate some logging
-	w.WriteByte(35)
+	defaultLogWriter.(*logWriter).w.WriteByte(35)
```

Comment:
these don't really add much to the tests. Each one should probably be creating its own LogWriter instance.